ml_ops.sh

```diff
@@ -59,77 +59,47 @@ LDA_OUTPUT_DIR=${DSOURCE}/${FDATE}
 
 TOPIC_COUNT=20
 
+nodes=${NODES[0]}
+for n in "${NODES[@]:1}" ; do nodes+=",${n}"; done
+
 hdfs dfs -rm -R -f ${HDFS_WORDCOUNTS}
 wait
 
 mkdir -p ${LPATH}
 rm -f ${LPATH}/*.{dat,beta,gamma,other,pkl} # protect the flow_scores.csv file
 
-
-time spark-submit --class "org.opennetworkinsight.Dispatcher" --master yarn-client --executor-memory  ${SPK_EXEC_MEM} \
-  --driver-memory 2g --num-executors ${SPK_EXEC} --executor-cores 1 --conf spark.shuffle.io.preferDirectBufs=false \
-   --conf shuffle.service.enabled=true --conf spark.driver.maxResultSize="2g"   target/scala-2.10/oni-ml-assembly-1.1.jar \
-   ${PREPROCESS_STEP} -i ${RAWDATA_PATH} -o ${HDFS_WORDCOUNTS} -f ${FEEDBACK_PATH} -d ${DUPFACTOR}
-
-wait
-
-hdfs dfs -copyToLocal  ${HDFS_WORDCOUNTS}/part-* ${LPATH}/.
-wait
-
-cd ${LPATH}
-cat part-* > doc_wc.dat
-rm -f part-*
-
-#   lda  stage
-
-cd ${LUSER}/ml
-time python lda_pre.py ${LPATH}/
-rm -f ${LPATH}/doc_wc.dat
-wait
-
-#  copy input files to all nodes in a new folder for this day's LDA results
-
-for d in "${NODES[@]}" 
-do 
-	echo "copying $d ${LPATH} ${LUSER}"
-	ssh $d 'mkdir '"'${LUSER}'"'/ml/'"'${DSOURCE}'"
-	scp -r ${LPATH} $d:${LUSER}/ml/${DSOURCE}/.
-done
-wait
-
-cd ${LDAPATH}
-
-${MPI_PREP_CMD}
-time ${MPI_CMD} -n ${PROCESS_COUNT} -f machinefile ./lda est 2.5 ${TOPIC_COUNT} settings.txt \
-  ${PROCESS_COUNT} ../${LDA_OUTPUT_DIR}/model.dat random ../${LDA_OUTPUT_DIR}
-wait
-
-
-cd ${LUSER}/ml
-time python lda_post.py ${LPATH}/
-
-# post-processing stage
-
-hdfs dfs -rm -R -f ${HDFS_DOCRESULTS}
-hdfs dfs -put ${LOCAL_DOCRESULTS} ${HDFS_DOCRESULTS}
-
-hdfs dfs -rm -R -f ${HDFS_WORDRESULTS}
-hdfs dfs -put ${LOCAL_WORDRESULTS} ${HDFS_WORDRESULTS}
-
-hdfs dfs -rm -R -f ${HDFS_WORDCOUNTS}
 hdfs dfs -rm -R -f ${HDFS_SCORED_CONNECTS}
 
-#kinit -kt /etc/security/keytabs/smokeuser.headless.keytab <user-id>
-wait
-time spark-submit --class "org.opennetworkinsight.Dispatcher" --master yarn-client --executor-memory  ${SPK_EXEC_MEM} \
-  --driver-memory 2g --num-executors ${SPK_EXEC} --executor-cores 1 --conf spark.shuffle.io.preferDirectBufs=false \
-  --conf shuffle.service.enabled=true --conf spark.driver.maxResultSize="2g" \
-  target/scala-2.10/oni-ml-assembly-1.1.jar ${POSTPROCESS_STEP} -i ${RAWDATA_PATH} -d ${HDFS_DOCRESULTS} \
-  -w ${HDFS_WORDRESULTS} -s ${HDFS_SCORED_CONNECTS} -t ${TOL}
+# Add -p <command> to execute pre MPI command.
+# Pre MPI command can be configured in /etc/duxbay.conf
+# In this script, after the line after -c ${MPI_CMD} add:
+# -p ${MPI_PREP_CMD}
+
+
+time spark-submit --class "org.opennetworkinsight.LDA" --master yarn-client --executor-memory  ${SPK_EXEC_MEM} \
+  --driver-memory 2g --num-executors ${SPK_EXEC} --executor-cores 1 --conf spark.shuffle.io.preferDirectBufs=false    \
+  --conf shuffle.service.enabled=true --conf spark.driver.maxResultSize="2g" target/scala-2.10/oni-ml-assembly-1.1.jar \
+   ${DSOURCE} \
```

Comment:
might want to use longer, more explanatory  input flags... eg. --input --dupfactor etc.... the option builder should have those already